src/lib/y2partitioner/actions/controllers/lvm_vg.rb

```diff
@@ -0,0 +1,364 @@
+            raise ArgumentError,
+              "The device #{device} is not used as physical volumen by the volume group #{vg_name}"
+          end
+
+          # TODO: restore status and descendants of the device when it makes sense
+          vg.remove_lvm_pv(device)
+        end
+
+      private
+
+        # Current devicegraph
+        #
+        # @return [Y2Storage::Devicegraph]
+        def working_graph
+          DeviceGraphs.instance.current
+        end
+
+        # Sets initial values
+        #
+        # @note A default volume group name and extent size is assigned, and a new volume
+        # group is created.
+        def initialize_values
+          @vg_name = DEFAULT_VG_NAME
+          @extent_size = DEFAULT_EXTENT_SIZE
+          @vg = Y2Storage::LvmVg.create(working_graph, vg_name)
+        end
+
+        # Checks whether the given volume group name is empty
+        #
+        # @return [Boolean] true if volume group name is "" or nil; false otherwise
+        def empty_vg_name?
+          vg_name.nil? || vg_name.empty?
+        end
+
+        # Error message to show when the given volume group name is empty
+        #
+        # @return [String]
+        def empty_vg_name_message
+          _("Enter a name for the volume group.")
+        end
+
+        # Checks whether the given volume group name has illegal characters
+        #
+        # @see ALLOWED_NAME_CHARS
+        #
+        # @return [Boolean] true if volume group name has illegal characters;
+        #   false otherwise
+        def illegal_vg_name?
+          vg_name.split(//).any? { |c| !ALLOWED_NAME_CHARS.include?(c) }
+        end
+
+        # Error message to show when the given volume group name contains illegal
+        # characters
+        #
+        # @return [String]
+        def illegal_vg_name_message
+          _("The name for the volume group contains illegal characters. Allowed\n" \
+            "are alphanumeric characters, \".\", \"_\", \"-\" and \"+\"")
+        end
+
+        # Checks whether it exists another device with the given volume group name
+        #
+        # @return [Boolean] true if there is another device with the same name;
+        #   false otherwise
+        def duplicated_vg_name?
+          # libstorage has the logic to generate a volume group name, so the name could
+          # assigned to the volume group, then check the duplicity of the name and restore
+          # previous name if the given one is duplicated. Instead of that, the logic to
+          # generate the volume group name has been duplicated here due to its simplicity.
+          name = File.join("/dev", vg_name)
+          Y2Storage::BlkDevice.all(working_graph).map(&:name).include?(name)
+        end
+
+        # Error message to show when the given volume group name is duplicated
+        #
+        # @return [String]
+        def duplicated_vg_name_message
+          # TRANSLATORS: vg_name is replaced by the name of the volume group
```

Comment:
NP: I would say: `%{vg_name} is replaced...` The whole point is that translators may not be familiar with the `%{whatever}` syntax.